pkg/aws/cluster/validate_cluster.go

```diff
@@ -83,6 +88,28 @@ func (v *Validator) ClusterLabelValuesValid(oldCluster *capiv1alpha2.Cluster, ne
 	return aws.ValidateLabelValues(&aws.Handler{K8sClient: v.k8sClient, Logger: v.logger}, oldCluster, newCluster)
 }
 
+func (v *Validator) ReleaseVersionValid(oldCluster *capiv1alpha2.Cluster, newCluster *capiv1alpha2.Cluster) error {
+	var err error
+
+	if key.Release(newCluster) == key.Release(oldCluster) {
+		return nil
+	}
+	// Retrieve the `Release` CR.
+	releaseVersion, err := aws.ReleaseVersion(newCluster, []mutator.PatchOperation{})
+	if err != nil {
+		return microerror.Maskf(parsingFailedError, "unable to parse release version from Cluster")
+	}
+	release, err := aws.FetchRelease(&aws.Handler{K8sClient: v.k8sClient, Logger: v.logger}, releaseVersion)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+	if release.Spec.State == "deprecated" {
```

Comment:
Do we consider user permissions? Can Giant Swarm staff still upgrade to deprecated releases?

Reply:
I am still testing but will ensure that works.

Reply:
Yes, you can still edit labels freely. This affects only customers and things like the giant swarm API. The only change to how it was with gatekeeper is that customers can also upgrade the cluster by changing the version label.